### ISSUE TITLE
Maximum redshift of quasar

### ISSUE THREAD

**Issue by londumas** [desihub/redrock#122]
The current redshift distribution:
https://desi.lbl.gov/svn/code/desimodel/trunk/data/targets/nz_qso.dat
is not zero at z>4., currently redrock stops at z=4. Should we fit up to z=5?
What do you think @nathalieGit and @sbailey?

![distribution_redshift](https://user-images.githubusercontent.com/3166436/41363281-bfbea150-6ef0-11e8-8c65-c7b5bd30a51f.png)

**Comment by nathalieGit** [comment]
It’d be nice indeed to fit to z=5. The 1D analysis for instance does use quasars till as high as are available.


> On 13 Jun 2018, at 18:03, Hélion du Mas des Bourboux <notifications@github.com> wrote:
> 
> The current redshift distribution:
> https://desi.lbl.gov/svn/code/desimodel/trunk/data/targets/nz_qso.dat
> is not zero at z>4., currently redrock stops at z=4. Should we fit up to z=5?
> What do you think @nathalieGit and @sbailey?
> 
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 

-----------------------------------------------------------------------------
Nathalie Palanque-Delabrouille

Irfu/DPhP cosmology group
CEA, Université Paris-Saclay, 91191 Gif sur Yvette, France
tel: (+33) 1 69 08 39 62   
-------------------------------------------------------------------------------

**Comment by andreufont** [comment]
Hi Helion,

On 13 June 2018 at 17:03, Hélion du Mas des Bourboux <
notifications@github.com> wrote:

> The current redshift distribution:
> https://desi.lbl.gov/svn/code/desimodel/trunk/data/targets/nz_qso.dat
> is not zero at z>4., currently redrock stops at z=4. Should we fit up to
> z=5?
> What do you think @nathalieGit <https://github.com/nathalieGit> and
> @sbailey <https://github.com/sbailey>?
>
It depends on what we want to use redrock for. If we want to use it for
something equivalent to the SDSS "quasar catalog", I'd say we want Redrock
to be able to fit redshifts all the way to z=7 or so... We don't want to
miss the most attractive quasars!

If we just want something for BAO + P1D analyses, then it is probably good
enough to stop at z=5.5 or so, although even then I would say that a quasar
at z=5.6 would be pretty cool for P1D.

Cheers,
Andreu

**Comment by londumas** [comment]
Looking with the current templates, we can go up to `z=6`, not up to `z=7`, simply because the templates don't go to enough low lambda_RF.
Comparing redrock+archetypes to DR12_Superset as a truth table, we improve slightly the number of good qso from `94.08%` up to `95.29%` and reduces the number of failures from `3.53%` to `2.81%`. We also improve the number of `ZWARN==0` qso from `108,288` to `108,819` (i.e. `531` qso).

zqso_max = 4
```
                   ntarg   good  fail  miss  lost
GALAXY              6669  82.40 15.58  0.36  1.66
QSO               110936  94.08  3.53  0.80  1.59
STAR               81657  88.59  2.41  5.46  3.53

                         | GALAXY | QSO    | STAR   | 
GALAXY              6534 |  93.79 |   4.76 |   1.45 | 
QSO               108288 |   1.68 |  98.21 |   0.10 | 
STAR               74310 |   2.10 |   0.60 |  97.30 | 
```

zqso_max = 6
```
                   ntarg   good  fail  miss  lost
GALAXY              6669  82.40 15.58  0.36  1.66
QSO               110936  95.29  2.81  0.57  1.34
STAR               81657  88.60  2.41  5.46  3.54

                         | GALAXY | QSO    | STAR   | 
GALAXY              6534 |  93.79 |   4.76 |   1.45 | 
QSO               108819 |   1.27 |  98.63 |   0.10 | 
STAR               74311 |   2.10 |   0.60 |  97.30 | 
```

**Comment by moustakas** [comment]
The maximum redshift accommodated by the SIMQSO templates should now be `z=7`; see https://github.com/desihub/desisim/issues/381#issuecomment-400870248.

**Comment by sbailey** [comment]
fixed by #130
